### PR TITLE
fix: throw error on NaN Integer type ENV variables

### PR DESCRIPTION
**Pull request by kimpers**
# Description

Noticed some issues with missing Integer type env variables. `parseInt(undefined, 10` returns `NaN` instead of throwing, which makes the check not error out even if the value is missing.

# Testing Instructions

# Checklist

-   [ ] Update [documentation](https://github.com/0xProject/website/blob/development/mdx/api/index.mdx) as needed. **Website Documentation PR:**
-   [ ] Prefix PR title with `[WIP]` if necessary.
-   [ ] Add tests to cover changes as needed.

### CODE REVIEW

File: src/config.ts [hunk 1]
```diff
@@ -507,6 +507,9 @@ function assertEnvVarType(name: string, value: any, expectedType: EnvVarType): a
         case EnvVarType.Integer:
             try {
                 returnValue = parseInt(value, 10);
+                if (isNaN(returnValue)) {
+                    throw new Error();
```
Comment (dekz) [inline src/config.ts:511]:
```suggestion
                    throw new Error(`${name} must be a valid integer, found ${value}.`);
```

Comment (kimpers) [reply src/config.ts:511]: It's wrapped in a catch below which throws with an appropriate message, I don't think the message here has any impact.

**Comment by opaolini** [comment]
I think we should just have:
```
        case EnvVarType.Integer:
                returnValue = parseInt(value, 10);
                if (isNaN(returnValue)) {
                    throw new Error(`${name} must be a valid integer, found ${value}.`);
                }
            return returnValue;
```

And maybe just have the:
```
        case EnvVarType.KeepAliveTimeout:
            try {
                returnValue = parseInt(value, 10);
            } catch (err) {
                throw new Error(`${name} must be a valid integer, found ${value}.`);
            }
            return returnValue;
        case EnvVarType.ChainId:
            try {
                returnValue = parseInt(value, 10);
            } catch (err) {
                throw new Error(`${name} must be a valid integer, found ${value}.`);
            }
            return returnValue;
```

As
```
        case EnvVarType.Integer:
        case EnvVarType.KeepAliveTimeout:
        case EnvVarType.ChainId:
                returnValue = parseInt(value, 10);
                if (isNaN(returnValue)) {
                    throw new Error(`${name} must be a valid integer, found ${value}.`);
                }
            return returnValue;
```

**Review by opaolini — APPROVED** [review]

**Comment by kimpers** [comment]
deploy staging

**Comment by kimpers** [comment]
deploy production

**Pull request merged**